### PR TITLE
set proper ACLs for device during own

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: local/core/getOwnership.go [hunk 1]
```diff
@@ -3,76 +3,20 @@ package core
...
-	}
-	err = h.Err()
-	if err != nil {
-		return schema.Doxm{}, err
+func (d *Device) GetOwnership(ctx context.Context, links schema.ResourceLinks) (schema.Doxm, error) {
+	ownLink, ok := links.GetResourceLink(schema.DoxmHref)
+	if !ok {
+		return schema.Doxm{}, fmt.Errorf("not found")
 	}
+	getOwnlink := ownLink
```
Comment (der-dano) [inline local/core/getOwnership.go:16]: isn't this just a reference copy?

File: local/core/ownDevice.go [hunk 2]
```diff
@@ -201,6 +207,72 @@ func (d *Device) setACL(ctx context.Context, links schema.ResourceLinks, ownerID
...
+					{
+						Href:       "/oic/sec/sp",
+						Interfaces: []string{"*"},
+					},
+					{
+						Href:       "oic/sec/pstat",
+						Interfaces: []string{"*"},
+					},
+					{
+						Href:       "oic/sec/doxm",
```
Comment (der-dano) [inline local/core/ownDevice.go:227]: schema.DoxmHref ?

File: schema/link.go [hunk 3]
```diff
@@ -124,6 +124,38 @@ func (r ResourceLink) GetEndpoints() []Endpoint {
...
+		switch addr.GetScheme() {
+		case string(TCPSecureScheme), string(UDPSecureScheme):
+			endpoints = append(endpoints, ep)
+		}
+	}
+	return endpoints
+}
+
+// GetUnsecureEndpoints returns unsecure endpoints in order of priority.
+func (r ResourceLink) GetUnsecureEndpoints() []Endpoint {
```
Comment (der-dano) [inline schema/link.go:144]: Is it neccessary (required by some API) to have a separate functions for secure and unsecure? Couldn't it be unified into a single function with a (un)secure parameter?

**Review by der-dano — COMMENTED** [review]
Just a couple of points...

File: local/core/getOwnership.go (hunk 1 above)
Comment (jkralik) [reply local/core/getOwnership.go:16]: ownLink is a `struct` so it is a copy of struct like in `C`

File: local/core/ownDevice.go (hunk 2 above)
Comment (jkralik) [reply local/core/ownDevice.go:227]: Fixed

File: schema/link.go (hunk 3 above)
Comment (jkralik) [reply schema/link.go:153]: I refactored to use common function.

**Review by der-dano — APPROVED** [review]

**Pull request merged**